core/src/main/java/hudson/model/Queue.java

```diff
@@ -1509,8 +1533,20 @@ public Api getApi() {
          * VM-wide unique ID that tracks the {@link Task} as it moves through different stages
```

Comment:
NIT: should this not be the "Jenkins master" wide - rather than the VM-Wide.
(VM-Wide implies on restart you could get clashes.)

What happens on JOC when scheduling across masters - if this is not suitable will we not have to invent yet another way?